porting/configure_test_fix/Overlay.rst

```diff
@@ -3,82 +3,150 @@
+
+- Files in directories containing none of the above special files are bind-mounted over files with an absolute path derived by stripping the above prefix. Subdirectories will be further traversed and evaluated according to the outlined rules.
+
+The specifics on how an overlay file or directory can be placed there depends on the used build system and will be described below.
+
+Example
+"""""""
+
+Contents of ``overlay/system/halium``::
 
-The first of these three files, 70-android.rules, is the one you created when bringing up Lomiri. You can extract this file from the device using ``adb pull`` or ``scp`` and copy it to the 'ubuntu' directory of your device source tree, making sure to add the corresponding line to your ``device.mk`` file, as described above.
+    overlay
+    └── system
+        └── halium
+            └── etc
+                ├── foo
+                │   ├── .halium-overlay-dir
+                │   ├── conf.d
+                │   │   ├── 50-drivers.conf
+                │   │   └── 90-local.conf
+                │   └── foo.conf
+                ├── bar
+                │   ├── .halium-override-dir
+                │   └── bar.conf
+                └── bazrc
 
-Explanation:
+Contents of the underlying filesystem image::
 
-The string before the colon '$(LOCAL_PATH)/ubuntu/70-android.rules' specifies the path to the source file to be injected. The string after the colon 'system/halium/lib/udev/rules.d/70-android.rules' specifies the target location on your device.
+    /etc
+    ├── foo
+    │   ├── .halium-overlay-dir
+    │   ├── conf.d
+    │   │   ├── 10-global.conf
+    │   │   ├── 20-system.conf
+    │   │   └── 90-local.conf
+    │   └── foo.conf
+    ├── bar
+    │   ├── .halium-override-dir
+    │   ├── bar.conf
+    │   └── baz.conf
+    └── quuxrc
 
-The general steps to follow are thus:
-    1. Copy the file you wish to modify to the 'ubuntu' directory you have created in your device source tree.
-    2. Edit the file as needed.
-    3. Add a line to the PRODUCT_COPY_FILES section of your device.mk file as shown above.
-    4. Rebuild your system.img and reflash together with the ubports rootfs.
+The resulting combined filesystem the looks as follows::
 
-.. Note::
+    /etc
+    ├── foo
+    │   ├── .halium-overlay-dir
+    │   ├── conf.d
+    │   │   ├── 10-global.conf
+    │   │   ├── 20-system.conf
+    │   │   ├── 50-drivers.conf (added)
+    │   │   └── 90-local.conf   (replaced)
+    │   └── foo.conf (replaced)
+    ├── bar (replaced)
+    │   ├── .halium-override-dir
+    │   └── bar.conf
+    └── quuxrc (replaced)
 
-    The target paths for the files mentioned above are *not* randomly chosen. You must use the specified paths. 
+The directory ``/etc/foo`` has been merged, whereas the ``/etc/bar`` directory and the file ``/etc/quuxrc`` have been replaced.
 
-.. Note::
+.. _Overlay-steps:
+
+Overlay files in full system.img builds
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
```suggestion
------------------------------------------
```

To make the heading the same size as "Understading mount-based overlay".

Reply:
OK, I've moved both sub-subsections up one level.